app/src/main/java/com/seafile/seadroid2/cameraupload/CameraSyncAdapter.java

```diff
@@ -500,46 +503,67 @@ private String varArgs(int count) {
      * @param cursor
      * @throws SeafException
      */
-    private void iterateCursor(SyncResult syncResult, DataManager dataManager, Cursor cursor) throws SeafException, InterruptedException {
+    private void iterateCursor(SyncResult syncResult, DataManager dataManager, Cursor cursor, String mediaId) throws SeafException, InterruptedException {
 
         tasksInProgress.clear();
 
+        File file;
+        Uri uri;
+        String id;
         // upload them one by one
         while (!isCancelled() && cursor.moveToNext()) {
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+                id = cursor.getString(cursor.getColumnIndexOrThrow(mediaId));
+                if (MediaStore.Images.Media._ID.equals(mediaId)) {
+                    uri = Uri.withAppendedPath(MediaStore.Images.Media.EXTERNAL_CONTENT_URI, id);
+                } else {
+                    uri = Uri.withAppendedPath(MediaStore.Video.Media.EXTERNAL_CONTENT_URI, id);
+                }
+                if (uri == null) {
+                    syncResult.stats.numSkippedEntries++;
+                    continue;
+                }
+                file = new File(Utils.getRealPathFromURI(SeadroidApplication.getAppContext(), uri, mediaId));
+            } else {
+                int dataColumn = cursor.getColumnIndexOrThrow(MediaStore.MediaColumns.DATA);
+                if (cursor.getString(dataColumn) == null) {
+                    syncResult.stats.numSkippedEntries++;
+                    continue;
+                }
+                file = new File(cursor.getString(dataColumn));
 
-            int dataColumn = cursor.getColumnIndexOrThrow(MediaStore.MediaColumns.DATA);
-            int bucketColumn = cursor.getColumnIndexOrThrow(MediaStore.Images.ImageColumns.BUCKET_DISPLAY_NAME);
-
-            // some inconsistency in the Media Provider? Ignore and continue
-            if (cursor.getString(dataColumn) == null) {
-                syncResult.stats.numSkippedEntries++;
-                continue;
             }
-
-            File file = new File(cursor.getString(dataColumn));
+            Utils.utilsLogInfo(true,"======iterateCursor");
+//            String id = cursor.getString(cursor.getColumnIndexOrThrow(MediaStore.Images.Media._ID));
+//            Uri uri = Uri.withAppendedPath(MediaStore.Images.Media.EXTERNAL_CONTENT_URI, id);
+//            file = new File(Utils.getRealPathFromUri(SeadroidApplication.getAppContext(), uri));
+            int bucketColumn = cursor.getColumnIndexOrThrow(MediaStore.Images.ImageColumns.BUCKET_DISPLAY_NAME);
             String bucketName = cursor.getString(bucketColumn);
 
             // local file does not exist. some inconsistency in the Media Provider? Ignore and continue
             if (!file.exists()) {
                 // Log.d(DEBUG_TAG, "Skipping media "+file+" because it doesn't exist");
+                Utils.utilsLogInfo(true,"=====Skipping media "+file+" because it doesn't exist");
```

Comment:
Utils.utilsLogInfo(true,"=====Skipping media "+file+" because it doesn't exist");   
改成 
Utils.utilsLogInfo(true, "=====Skipping media " + file + " because it doesn't exist");

Reply:
已修改